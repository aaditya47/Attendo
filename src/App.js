import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from 'react-router-dom';
import {
  ChakraProvider,
  Box,
  VStack,
  Grid,
  theme,
} from '@chakra-ui/react';
import { ColorModeSwitcher } from './ColorModeSwitcher';

import LoginSelector from './pages/LoginSelector';
import StudentLogin from './pages/StudentLogin';
import TeacherLogin from './pages/TeacherLogin';
import StudentDashboard from './pages/StudentDashboard';
import LeaveForm from './pages/LeaveForm';
import PastLeaves from './pages/PastLeaves';
import TeacherTimeTable from './pages/TeacherTimeTable';
import AttendanceUpdate from './pages/AttendanceUpdate';

function App() {
  return (
    <Router>
      <ChakraProvider theme={theme}>
        <Box textAlign="center" fontSize="xl">
          <Grid minH="100vh" p={3}>
            <ColorModeSwitcher justifySelf="flex-end" />
            <VStack spacing={8}>
              <Switch>
                <Route exact path="/" component={LoginSelector} />
                <Route exact path="/studentlogin" component={StudentLogin} />
                <Route exact path="/teacherlogin" component={TeacherLogin} />
                <Route exact path="/studentdashboard" component={StudentDashboard} />
                <Route exact path="/leaveform" component={LeaveForm} />
                <Route exact path="/pastleaves" component={PastLeaves} />
                <Route exact path="/teachertimetable" component={TeacherTimeTable} />
                <Route exact path="/attendanceupdate" component={AttendanceUpdate} />
              </Switch>
            </VStack>
          </Grid>
        </Box>
      </ChakraProvider>
    </Router>
  );
}

export default App;
