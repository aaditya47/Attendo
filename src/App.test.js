import renderer from 'react-test-renderer'
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App'
import {cleanup } from '@testing-library/react'

afterEach(cleanup)
it ("renders without crashing",()=>{
    const div=document.createElement("div");
    ReactDOM.render(<App></App>,div)
})
it("matches snapshot", () => {
    const tree = renderer.create(<App></App>).toJSON();
    expect(tree).toMatchSnapshot();
})