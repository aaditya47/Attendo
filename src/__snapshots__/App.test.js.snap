// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="css-12hvf5p"
  >
    <div
      className="css-1yryxfe"
    >
      <button
        aria-label="Switch to dark mode"
        className="chakra-button css-jsu1e6"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
          />
        </svg>
      </button>
      <div
        className="chakra-stack css-uc6l60"
      >
        <div
          className="css-mx5jal"
        >
          <div
            className="css-1jif775"
          >
            <div
              className="chakra-stack css-owjkmg"
            >
              <div
                className="css-0"
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "font-size": "2em",
                      "font-weight": "bold",
                    }
                  }
                >
                  <span
                    className="myTypist"
                  >
                    
                  </span>
                  <span
                    className="typist-cursor myCursor myTypist"
                    style={
                      Object {
                        "animation": "blink 700ms steps(1) infinite",
                        "animationDelay": "0ms",
                        "color": "",
                        "opacity": 0,
                        "visibility": "visible",
                      }
                    }
                  >
                    |
                  </span>
                </span>
              </div>
              <div
                className="css-8cmolj"
              >
                <div
                  className="css-xi606m"
                >
                  <h2
                    className="chakra-heading css-1dklj6k"
                  >
                    Welcome!
                  </h2>
                  <p
                    className="chakra-text css-rltemf"
                  >
                    Please select your login method
                  </p>
                </div>
                <div
                  className="css-1mdqppq"
                >
                  <button
                    className="chakra-button css-1b1shkm"
                    type="button"
                  >
                    <a
                      className="chakra-link css-f4h6uy"
                      href="/studentlogin"
                    >
                      Student
                    </a>
                  </button>
                </div>
                <div
                  className="css-1mdqppq"
                >
                  <button
                    className="chakra-button css-1b1shkm"
                    type="button"
                  >
                    <a
                      className="chakra-link css-f4h6uy"
                      href="/teacherlogin"
                    >
                      Teacher
                    </a>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            className="css-1fcwpvw"
            style={
              Object {
                "paddingTop": "20%",
              }
            }
          >
            <img
              className="chakra-image__placeholder css-0"
            />
          </div>
        </div>
      </div>
      <img
        alt="image"
        className="chakra-image__placeholder css-1591w8z"
      />
    </div>
  </div>,
  <span />,
]
`;
